feat(bigquery): Add example for creating a materialized view

ccarpentiere opened the pull request:
## Description

Fixes #269188404

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/bigquery/docs/materialized-views-create

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

ccarpentiere commented:
/gcbrun

glasnt commented:
@ccarpentiere Adding the `test.yml` with this sample means that the `terraform apply/destroy` won't be run. Is that intended? Can this sample be tested?

ccarpentiere commented:
I tested the sample with terraform apply with line 33 changed to `query = "SELECT global_id, faa_identifier, name, latitude, longitude FROM `bigquery-public-data.faa.us_airports`"`. I changed it to an example table to be more realistic - you wouldn't create a materialized view over a public dataset. However, terraform apply fails with that change introduced, because it tests whether the current user has access to the target table for the view, which doesn't exist. That's why I added test.yaml. Terraform fmt still works and was run against the sample.
